docs/DeveloperGuide.md

```diff
@@ -1,33 +1,372 @@
+fork the project to your own repository, and clone your fork into your working computer.    
+
+Now, Open Intellij 
+1. If you are not in the welcome screen, click `File` → `Close Project` to navigate to the starting project dialog)
+1. Set up the correct JDK version for Gradle.
+        1. Click `Configure` → `Project Structure for New Project`
+        1. Click `New...` and set it to the directory of the JDK
+1. Upon the successful configuration, you should be able to see the following screenshot:
+![JDK Config](images/JDK_Config_Screenshot.png)
+1. Lastly, **Import the project as a Gradle Project.**
+    * Intellij IDEA by default has the Gradle plugin installed. If you have disabled it, go to 
+    `File` → `Settings` → `Plugins` to re-enable them.
+    * Click Import Project (or Open or Import in newer version of Intellij).
+    * Locate the **build.gradle file** (not the root folder as you would do in a normal importing) and select it. Click `OK`.
+    * If asked, choose to `Open as Project` (not Open as File).
+    * Click `OK` to accept the default settings but do ensure that the selected version of Gradle JVM matches the 
+    JDK being used for the project.
+    * Wait for the importing process to finish (may take a few minutes, so get some :coffee: :relaxed:)
+1. **Verify the set up**:
+    * Run the `..main/java/seedu.duke/Duke`
+    * Upon a successful configuration, you should be able to see the welcome screen of the project as shown below:
+    ![verify_set_up](images/Verify_setup.PNG)
+    * Run some tests to ensure it is working properly.
+
+### Getting started
+
+1. **Configure the coding style**
+If using IDEA, follow the guide If using IDEA, follow the guide [se-edu/guides IDEA: Configuring the code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html) 
+to set up IDEA’s coding style to match ours.
+    :bulb: Tips: Optionally, you can follow the guide  [se-edu/guides Using Checkstyle](https://se-education.org/guides/tutorials/checkstyle.html) 
+    to find how to use the CheckStyle within IDEA e.g., to report problems as you write code.
+1. **Set up CI**
+This project comes with a GitHub Actions config files (in `.github/workflows` folder). When GitHub detects those files, 
+it will run the CI for your project automatically at each push to the master branch or to any PR. No set up required.
+1. **Learn the design**
+When you are ready to start coding, we recommend that you get some sense of the overall design by reading about [25HoursADay’s architecture](#architecture).
+
+## Design
+This section describes the design overview of the application.
+
+### Architecture
+
+The figure below shows the overall design of the application. Given below is a sequence diagram when adding a Todo task.
+<img src="https://raw.githubusercontent.com/AY2021S1-CS2113T-T12-2/tp/master/images/Overall_Architecture.JPG" alt="" width="300"/> <br/>
```

Comment:
Perhaps a shift the sequence diagram reference to after this image, as I was confused for a while, thinking its the figure being referenced.